db/migrations/20190715152452_create_single_flip_query.sql

```diff
@@ -19,6 +19,9 @@ CREATE TYPE api.flip_bid_snapshot AS
     updated      TIMESTAMP
 );
 
+COMMENT ON TYPE api.flip_bid_snapshot
+    IS E'Historical snapshots of the state of auctions on the Flip contracts, with data about associated Urn.';
```

Comment:
I found it pretty difficult to distinguish between `maker.flip` and `api.flip_bid_snaphot` in the docs, and there are similar ambiguities (for me) on the other auctions contracts as well as urns. Wondering whether part of it might be that we were originally only going to expose the `api` schema and are now planning to expose all. If so, maybe it's worthwhile to consolidate? Alternatively, if anyone with a better understanding can chime in I'd be happy to update the docs.

Reply:
Yeah you're exactly right; the `maker.flip/flap/flop` tables were meant to be sort of intermediary caches, not exposed to the client. The client-facing functions like `api.get_flip`, `api.all_flips` etc. (which return `snapshot`s) make use of those tables, and also do a couple more things at request-time, like determine whether the bid is `dealt`, and (in `flip`'s case) figure out which ilk and urn are associated with the bid. Also, since only the `api.-bid_snapshot` types were meant to be client-facing, they have "computed columns", while the ones in the `maker` schema don't.

I agree that it would probably make sense to do some consolidation. To that end, it seems feasible to remove all the request-time logic that `get_flip/flap/flop` are doing, and instead add the bids' `dealt`, `ilk`, and `urn` data to our trigger tables. Then we could probably trim the API down to just using a single type for each kind of bid.